canonical-web-and-design/maas-ui
# refactor(ui): refactor some common form Select components

## Done

- Created common form Select components that are used in a few places. Each of them do the same thing - load the relevant items into state and turn them into a select. These will also be used in the machine configuration tab (and likely devices and controllers)
  - `ArchitectureSelect`
  - `DomainSelect`
  - `MinimumKernelSelect`
  - `ResourcePoolSelect`
  - `ZoneSelect` 

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/machines/add and check that the selects still work correctly (i.e. is populated and can select option), and the form submits correctly

## Fixes

Fixes #2126
